grpc/estimator.proto

```diff
@@ -106,14 +120,33 @@ message SettingsResponse {
 // forces
 
 message Cost {
-  optional double total = 1;
-  optional double prior = 2;
-  optional double sensor = 3;
-  optional double force = 4;
-  optional double initial = 5;
+  double total = 1;
+  double prior = 2;
+  double sensor = 3;
+  double force = 4;
+  double initial = 5;
+  repeated double gradient = 6 [packed = true];
+  repeated double hessian = 7 [packed = true]; 
+  repeated double residual_prior = 8 [packed = true];
+  repeated double residual_sensor = 9 [packed = true];
+  repeated double residual_force = 10 [packed = true];
+  repeated double jacobian_prior = 11 [packed = true];
+  repeated double jacobian_sensor = 12 [packed = true];
+  repeated double jacobian_force = 13 [packed = true];
+  repeated double norm_gradient_sensor = 14 [packed = true];
+  repeated double norm_gradient_force = 15 [packed = true];
+  repeated double prior_matrix = 16 [packed = true];
+  repeated double norm_hessian_sensor = 17 [packed = true];
+  repeated double norm_hessian_force = 18 [packed = true];
+  int32 nvar = 19;
+  int32 nsensor = 20;
+  int32 nforce = 21;
 }
 
-message CostRequest {}
+message CostRequest {
+  optional bool derivatives = 1;
+  optional bool internals = 2;
+}
 
 message CostResponse {
   Cost cost = 1;
```

Comment:
Is the message Cost used anywhere else other than CostResponse?

Reply:
No. Should I just move it into `CostResponse`?